src/main/java/org/folio/rest/impl/UsersAPI.java

```diff
@@ -468,30 +483,100 @@ public void postUsersOutboxProcess(Map<String, String> okapiHeaders, Handler<Asy
       });
   }
 
-  private void updateUser(User entity, Map<String, String> okapiHeaders,
+  private void updateUser(User entity, Map<String, String> okapiHeaders, PostgresClient pgClient,
       Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
 
     Date now = new Date();
     entity.setCreatedDate(now);
     entity.setUpdatedDate(now);
-    PgUtil.put(TABLE_NAME_USERS, entity, entity.getId(), okapiHeaders, vertxContext, PutUsersByUserIdResponse.class, reply -> {
-      if (isDuplicateUsernameError(reply)) {
-        asyncResultHandler.handle(
-          succeededFuture(PutUsersByUserIdResponse
-            .respond400WithTextPlain(
-              "User with this username already exists")));
-        return;
-      }
-      if (isDuplicateBarcodeError(reply)) {
-        asyncResultHandler.handle(
-          succeededFuture(PutUsersByUserIdResponse
-            .respond400WithTextPlain(
-              "This barcode has already been taken")));
-        return;
-      }
-      logger.debug("Save successful");
-      asyncResultHandler.handle(reply);
-    });
+
+    pgClient.withTrans(conn -> usersService.getUserByIdForUpdate(conn, entity.getId())
+        .compose(userFromStorage -> {
+          if (userFromStorage == null) {
+            return succeededFuture(PutUsersByUserIdResponse.respond404WithTextPlain(entity.getId()));
+          }
+          boolean isConsortiaFieldsUpdated = isConsortiumUserFieldsUpdated(entity, userFromStorage);
```

Comment:
Basically we don't need this check: isConsortiumUserFieldsUpdated
We just implementing common domain event USER_UPDATED, mod-consortia should not know who will process these events and in future some another parts of the system can subscribe to the user domain events

Reply:
After internal discussion we agreed that we don't need to send USER_CREATED/USER_UPDATED/USER_DELETED for non consortia mode, because now payload contains user phone numbers and emails, and we need to exclude exposing this sensitive data as much as possible